src/batch/src/executor/project_set.rs

```diff
@@ -0,0 +1,291 @@
+            // output table
+            let mut projected_row_id_builder = I64ArrayBuilder::new(DEFAULT_CHUNK_BUFFER_SIZE);
+            let mut builders = data_types
+                .iter()
+                .map(|ty| ty.create_array_builder(DEFAULT_CHUNK_BUFFER_SIZE))
+                .collect_vec();
+
+            let results: Vec<_> = self
+                .select_list
+                .iter()
+                .map(|select_item| select_item.eval(&data_chunk))
+                .try_collect()?;
+
+            let mut lens = results
+                .iter()
+                .map(|result| for_both!(result, r=>r.len()))
+                .dedup();
+            let cardinality = lens.next().unwrap();
+            assert!(
+                lens.next().is_none(),
+                "ProjectSet has mismatched output cardinalities among select list."
+            );
+
+            // each iteration corresponds to the outputs of one input row
+            for row_idx in 0..cardinality {
+                let items = results
+                    .iter()
+                    .map(|result| match result {
+                        Either::Left(arrays) => Either::Left(arrays[row_idx].clone()),
+                        Either::Right(array) => Either::Right(array.value_at(row_idx)),
+                    })
+                    .collect_vec();
+                // The maximum length of the results of table functions will be the output length.
+                let max_tf_len = items
+                    .iter()
+                    .map(|i| i.as_ref().map_left(|arr| arr.len()).left_or(0))
+                    .max()
+                    .unwrap();
+
+                for i in 0..max_tf_len {
+                    projected_row_id_builder.append(Some(i as i64))?;
+                }
+
+                for (item, builder) in items.into_iter().zip_eq(builders.iter_mut()) {
+                    match item {
+                        Either::Left(array_ref) => {
+                            builder.append_array(&array_ref)?;
+                            for _ in 0..(max_tf_len - array_ref.len()) {
+                                builder.append_null()?;
+                            }
+                        }
+                        Either::Right(datum_ref) => {
+                            for _ in 0..max_tf_len {
+                                builder.append_datum_ref(datum_ref)?;
+                            }
+                        }
+                    }
+                }
+            }
+            let mut columns = Vec::with_capacity(self.select_list.len() + 1);
+            let projected_row_id: ArrayRef = Arc::new(projected_row_id_builder.finish()?.into());
+            let cardinality = projected_row_id.len();
+            columns.push(Column::new(projected_row_id));
+            for builder in builders {
+                columns.push(Column::new(Arc::new(builder.finish()?)))
+            }
+
+            let chunk = DataChunk::new(columns, cardinality);
```

Comment:
better to produce a large chunk?

Reply:
It's already a large chunk?

Reply:
Oh I see. I remembered that we have something like `DataChunkBuilder`, which will return an Option every time there's an insert. We can do something like:

```
if let Some(chunk) = builder.append(...) {
  yield chunk;
}
```

Is it used in anywhere else in RisingWave? We can reuse it and do split chunk.

Reply:
It seems we only have `StreamChunkBuilder` 🥸

Reply:
Oh, and there's `DataChunk::rechunk`

Reply:
Let's do it later